catalog/app/utils/json-schema.js

```diff
@@ -0,0 +1,84 @@
+import * as R from 'ramda'
+import isArray from 'lodash/isArray'
+import isNumber from 'lodash/isNumber'
+import isObject from 'lodash/isObject'
+import isString from 'lodash/isString'
+import isUndefined from 'lodash/isUndefined'
+
+function isSchemaArray(optSchema) {
+  return R.prop('type', optSchema) === 'array'
```

Comment:
we can make this significantly less verbose: `const isSchemaArray = R.propEq('type', 'array')`

Reply:
This is some inconsistency in Ramda. `R.prop` is safe for any value, `R.propEq` throws error when there is no object.
I need `undefined` here, and highlighted this with `opt` prefix

Reply:
i see